### PR TITLE
08 11 combined patch delete users topic query

### PR DESCRIPTION
**Pull request by OpusLearning**
Hi, this is a combined PR for tasks 8-11

### CODE REVIEW

File: controllers/articles.js [hunk 1]
```diff
@@ -2,50 +2,57 @@ const {
...
+
+exports.updateArticleById = async (req, res, next) => {
+  const { article_id } = req.params;
+  const { inc_votes } = req.body;
+  if (typeof inc_votes !== "number") {
+    return next({ status: 400, msg: "Bad request" });
+  }
+
+  try {
+    await selectArticleById(article_id);
```
Comment (Harrison365) [inline controllers/articles.js:52]: Not needed

File: models/comments.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+};
+
+exports.removeCommentById = (comment_id) => {
+  return db
+    .query("DELETE FROM comments WHERE comment_id = $1 RETURNING *;", [
+      comment_id,
+    ])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Comment not found" });
```
Comment (Harrison365) [inline models/comments.js:65]: Don't doe destructure. --> result
if(result.rowCount === 0) --> error

File: models/users.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const db = require("../db/connection");
+
+exports.fetchUsers = () => {
+  return db
+    .query("SELECT username, name, avatar_url FROM users;")
```
Comment (Harrison365) [inline models/users.js:5]: *

Comment (Harrison365) [inline models/users.js:5]: "*"

File: __tests__/master.test.js [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+
+  test("200: responds with articles  by the topic value specified in the query", () => {
+    return request(app)
+      .get("/api/articles?topic=cats")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeInstanceOf(Array);
+        expect(body.articles.length).toBeGreaterThan(0);
+        body.articles.forEach((article) => {
+          expect(article.topic).toBe("cats");
```
Comment (Harrison365) [inline __tests__/master.test.js:390]: Check all properties.

File: __tests__/master.test.js [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+  });
+
+  test("200: responds with an empty array when the topic does not exist", () => {
+    return request(app)
+      .get("/api/articles?topic=not-a-topic")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeInstanceOf(Array);
+        expect(body.articles.length).toBe(0);
+      });
```
Comment (Harrison365) [inline __tests__/master.test.js:402]: 404 error not 200

Comment (Harrison365) [reply __tests__/master.test.js:402]: We want an empty array if the topic exists but has no articles. If topic doesn't exist, throw error.

File: __tests__/master.test.js [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+  });
+
+  test("400: responds with an error for invalid queries", () => {
+    return request(app)
+      .get("/api/articles?sort_by=not-a-column")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
```
Comment (Harrison365) [inline __tests__/master.test.js:412]: Change this to your 404 topic=banana

File: models/comments.js (hunk 2 above)
Comment (OpusLearning) [reply models/comments.js:65]: I've now updated to rowCount thanks for the advice

File: models/users.js (hunk 3 above)
Comment (OpusLearning) [reply models/users.js:5]: updated query to *

File: __tests__/master.test.js (hunk 4 above)
Comment (OpusLearning) [reply __tests__/master.test.js:390]: updated to check for properties

**Pull request merged**